components/campaign/SealTokenDialog.tsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import { StyleSheet, View } from 'react-native';
+import { Action, bindActionCreators, Dispatch } from 'redux';
+import { connect } from 'react-redux';
+import { flatMap, keys, map, range, sortBy } from 'lodash';
+import { EventSubscription, Navigation } from 'react-native-navigation';
+import { t } from 'ttag';
+
+import { NavigationProps } from '../types';
+import { updateCampaign } from './actions';
+import { iconsMap } from '../../app/NavIcons';
+import { COLORS } from '../../styles/colors';
+import { AppState, getCampaign } from '../../reducers';
+import { CHAOS_TOKEN_ORDER, ChaosBag, ChaosTokenType } from '../../constants';
+import { Campaign, ChaosBagResults, NEW_CHAOS_BAG_RESULTS } from '../../actions/types';
+import SealTokenButton from './SealTokenButton';
+
+const uuidv4 = require('uuid/v4');
+
+export interface SealTokenDialogProps {
+  campaignId: number;
+}
+
+interface ReduxProps {
+  chaosBag: ChaosBag;
+  chaosBagResults: ChaosBagResults;
+}
+
+interface ReduxActionProps {
+  updateCampaign: (id: number, chaosBagResults: Partial<Campaign>) => void;
+}
+
+type Props = NavigationProps & SealTokenDialogProps & ReduxProps & ReduxActionProps;
+
+class SealTokenDialog extends React.Component<Props> {
+  static options() {
+    return {
+      topBar: {
+        title: {
+          text: t`Choose a Token`,
```

Comment:
I think title here should be Choose tokens to seal, and maybe the `X`.

Also a 'Done' button at bottom of tokens that just closes the dialog would be nice I think, easier than closing out from the top corner.  Can just use the stock react-native Button